components/shell/ConfigurationContext.tsx

```diff
@@ -63,6 +64,13 @@ export const ConfigurationProvider = ({
     isLoading: isServerConfigLoading,
   } = useServerConfigQuery(labelerAgent)
 
+  const appviewAgent = useMemo<Agent | null>(() => {
+    if (serverConfig?.appview) {
+      return new Agent(new CredentialSession(new URL(serverConfig.appview)))
+    }
+    return null
+  }, [serverConfig?.appview])
+
```

Comment:
The `ConfigurationProvider` logic is already pretty complex on its own. I thing that keeping this logic in its own hook (in another file even) would make things much easier to maintain.

Something like so (didn't try):

```ts
export function useAppviewAgent() {
  const { appview } = useConfigurationContext().serverConfig

  return useMemo<Agent | null>(() => {
    if (appview) {
      return new Agent(new CredentialSession(new URL(appview))))
    }
    return null
  }, [appview)])
}
```

Or, if you prefer a long lived agent instance (instead of one that gets created per component), use an `AppviewAgentProvider` (I don't think this is necessary though).

Reply:
Also note: If you are not going to use session manageement here, just do:
```ts
new Agent({ service: appview })
```

Reply:
I thought it would be nicer to keep the definition of pds and appview agent in the same place but not opposed to splitting them up.
the `new Agent()` instantiation doesn't work without a `SessionManager` instance so I'm not sure how I'd do `new Agent(appview)`? but yeah, agreed that the instantiation should be simpler if I don't need session management, if the api allowed it.